apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -1539,6 +1539,61 @@ static void tc_pthread_pthread_setcanceltype(void)
 }
 #endif
 
+#ifndef CONFIG_PTHREAD_MUTEX_UNSAFE
+static void *test_mutex_consistent(void *param)
+{
+	int ret = -1;
+	ret = pthread_mutex_lock(&g_mutex);
+	return (void *)ret;
+}
+
+/**
+* @fn                   :tc_pthread_pthread_mutex_consistent
+* @brief                :This tc tests pthread_mutex_consistent()
+* @Scenario             :the pthread_mutex_consistent() function can be used to mark the state protected by the mutex referenced by mutex as consistent again
+*                        and this situation arises, If an owner of a robust mutex terminates while holding the mutex,
+*                        the mutex becomes inconsistent. In this case, the mutex does not become normally usable again until the state is marked consistent.
+*                        after marking consistent next thread can acquire the lock but will receive EOWNERDEAD.
+* @API'scovered         :pthread_mutex_consistent
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+static void tc_pthread_pthread_mutex_consistent(void)
+{
+	int ret;
+	void *status;
+	pthread_t thread1;
+	pthread_t thread2;
+	pthread_mutexattr_t mutex_attr;
+
+	pthread_mutexattr_init(&mutex_attr);
+	pthread_mutexattr_setrobust(&mutex_attr, PTHREAD_MUTEX_ROBUST);
+	pthread_mutex_init(&g_mutex, &mutex_attr);
+
+	/* creates the 1st thread to lock mutex */
+	ret = pthread_create(&thread1, NULL, &test_mutex_consistent, NULL);
+	TC_ASSERT_EQ_CLEANUP("pthread_create", ret, 0, pthread_mutex_unlock(&g_mutex));
+	pthread_join(thread1, &status);
+	TC_ASSERT_EQ_CLEANUP("pthread_mutex_consistent", (int)status, 0, pthread_mutex_unlock(&g_mutex));
+
+	/* make the mutex consistent, which became inconsistent because holding thread is dead*/
+	ret = pthread_mutex_consistent(&g_mutex);
+	TC_ASSERT_EQ_CLEANUP("pthread_mutex_consistent", ret, 0, pthread_mutex_unlock(&g_mutex));
+
+	/* create 2nd thread to lock the mutex again, which is consistent now and try to acquire the lock, it should be succeed and should return EOWNERDEAD*/
+	ret = pthread_create(&thread2, NULL, &test_mutex_consistent, NULL);
+	TC_ASSERT_EQ_CLEANUP("pthread_create", ret, 0, pthread_mutex_unlock(&g_mutex));
```

Comment:
Is this necessary even you successfully call pthread_mutex_consistent in this case?

Reply:
I think it can be better to check for the return value from pthread_create.
what you think, should I remove it or keep it?

Reply:
Oh sorry. I missed exact word.
Is `pthread_mutex_unlock(&g_mutex)` necessary even you successfully call pthread_mutex_consistent in this case?
I don't mean that line itself.
What I said is
TC_ASSERT_EQ_CLEANUP -> TC_ASSERT_EQ.
Because unlock looks unnecessary.

Reply:
I agree checking the return value from pthread_create.

Reply:
I think `pthread_mutex_unlock` is needed. Because 2nd thread took the mutex.
`g_mutex` is used from another testcase, so It is better to unlock.

Reply:
`pthread_mutex_unlock` in this line is executed when 2nd pthread_create is failed. It means the mutex is not locked. That's why I think it's unnecessary.

Reply:
yes sunghan, you are right.
i have made the required changes.